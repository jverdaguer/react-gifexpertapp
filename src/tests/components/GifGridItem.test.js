import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

const title = 'My Title';
const url = 'https://localhost/my_image.jpg';
const wrapper = shallow(<GifGridItem title={title} url={url} />);

describe('Test on GifGridItems', () => {
    test('should mount Component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should <p> have a title match', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('should have an image with src and alt', () => {
        const p = wrapper.find('img');
        const image = p.props();
        expect(image.src.trim()).toBe(url);
        expect(image.alt.trim()).toBe(title);
    });

    test('should have CSS class animate__fadeIn', () => {
        const div = wrapper.find('div');
        // expect(div.hasClass('animate__fadeIn'));
        expect(div.hasClass('animate__fadeIn')).toBe(true);
    });
});
