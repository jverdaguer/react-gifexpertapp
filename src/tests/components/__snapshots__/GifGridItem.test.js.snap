// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGridItems should mount Component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="My Title"
    src="https://localhost/my_image.jpg"
  />
  <p>
    My Title
  </p>
</div>
`;
